KsApi/Service+DecodeHelpers.swift

```diff
@@ -0,0 +1,102 @@
+import Argo
+import Foundation
+import Prelude
+import ReactiveExtensions
+import ReactiveSwift
+
+extension Service {
+  func decodeModel<M: Argo.Decodable>(_ json: Any) ->
+    SignalProducer<M, ErrorEnvelope> where M == M.DecodedType {
+    return SignalProducer(value: json)
+      .map { json in decode(json) as Decoded<M> }
+      .flatMap(.concat) { (decoded: Decoded<M>) -> SignalProducer<M, ErrorEnvelope> in
+        switch decoded {
+        case let .success(value):
+          return .init(value: value)
+        case let .failure(error):
+          print("Argo decoding model \(M.self) error: \(error)")
+          return .init(error: .couldNotDecodeJSON(error))
+        }
+      }
+  }
+
+  func decodeModels<M: Argo.Decodable>(_ json: Any)
+    -> SignalProducer<[M], ErrorEnvelope> where M == M.DecodedType {
+    return SignalProducer(value: json)
+      .map { json in decode(json) as Decoded<[M]> }
+      .flatMap(.concat) { (decoded: Decoded<[M]>) -> SignalProducer<[M], ErrorEnvelope> in
+        switch decoded {
+        case let .success(value):
+          return .init(value: value)
+        case let .failure(error):
+          print("Argo decoding model error: \(error)")
+          return .init(error: .couldNotDecodeJSON(error))
+        }
+      }
+  }
+
+  func decodeModel<M: Argo.Decodable>(_ json: Any) ->
+    SignalProducer<M?, ErrorEnvelope> where M == M.DecodedType {
+    return SignalProducer(value: json)
+      .map { json in decode(json) as M? }
+  }
+
+  // MARK: Swift.Codable
```

Comment:
I think Xcode won't register this comment unless it is formatted like this `// MARK: -`